kalitectl.py

```diff
@@ -663,12 +663,17 @@ def docopt(doc, argv=None, help=True, version=None, options_first=False):  # @Re
     
     # if matched and left == []:  # better error message if left?
     if collected:  # better error message if left?
+        
         result = Dict((a.name, a.value) for a in (pattern.flat() + collected))
-        collected_django_options = len(result.get('DJANGO_OPTIONS', []))
-        result['DJANGO_OPTIONS'] = (
-            result.get('DJANGO_OPTIONS', []) +
-            sys.argv[len(collected) + (collected_django_options or 1):]
-        )
+        
+        # count the number of arguments that were not Django options, and then properly read in Django
+        # options from sys.argv (we can't get them from "DJANGO_OPTIONS", since that doesn't catch 
+        # long-form options like "--git-migrate")
+        kalite_command_arg_count = len(collected)
+        if "DJANGO_OPTIONS" in result:
+            kalite_command_arg_count -= 1
+        result['DJANGO_OPTIONS'] = sys.argv[kalite_command_arg_count+1:]
+        
```

Comment:
@benjaoming The old code here didn't work when mixing argument types for Django options (long-form-args like "--git-migrate" were not caught in "collected", but left in "__left", and the indexing counts were off).

Let me know if you see any issues with the way I've structured it now. It just ignores any stuff matched by `DJANGO_OPTIONS` and then pulls all the Django options in from `sys.argv`.